asnake/client/web_client.py

```diff
@@ -86,14 +88,32 @@ def __init__(self, **config):
         log.debug("client created")
 
 
-    def authorize(self, username=None, password=None):
+    def authorize(self, username=None, password=None, session_token=None, session_header_name=None):
         '''Authorizes the client against the configured archivesspace instance.
 
         Parses the JSON response, and stores the returned session token in the session.headers for future requests.
         Asks for a "non-expiring" session, which isn't truly immortal, just long-lived.'''
 
-        username = username or self.config['username']
-        password = password or self.config['password']
+        # Populate values from config if empty
+        username      = username or self.config.get('username', None)
+        password      = password or self.config.get('password', None)
+        session_token = session_token or self.config.get('session_token', None)
+
+        if all((username, password, session_token,)):
+            log.debug('argument error in authorize')
+            raise ASnakeAuthError("Cannot set both username/password and session_token")
+
+        session_header_name = session_header_name or self.config['session_header_name']
```

Comment:
Should this be `self.config.get('session_header_name', None)`?

Reply:
It doesn't need to be, imo, because a default is provided in the configurator - this can't be None unless it's explicitly set none by the user, in which case an error SHOULD be thrown.